BasicCoin.sol

```
@@ -121,3 +165,95 @@ contract BasicCoin is Owned, Token {
+contract BasicCoinManager is Owned {
+  // a structure wrapping a deployed BasicCoin
+  struct Deployed {
+    address coin;
+    address owner;
+    bool tokenreg;
+  }
+
+  // a new BasicCoin has been deployed
+  event Created(address indexed owner, address coin, bool tokenreg);
+
+  // a list of all the deployments
+  Deployed[] deployments;
+
+  // all BasciCoins for a specific owner
+  mapping (address => uint[]) ownedDeployments;
+
+  // the network registry contract
+  Registry registry;
+
+  // the name of TokenReg
+  bytes32 constant tokenregName = sha3('tokenreg');
+
+  // the base, tokens denoted in micros (matches up with BasicCoin above)
+  uint constant public base = 1000000;
+
+  // create the coin creator, storing the network registry
+  function BasicCoinManager(address _registryAddress) {
+    updateRegistry(_registryAddress);
+  }
+
+  // return the number of deployments
+  function count() constant returns (uint) {
+    return deployments.length;
+  }
+
+  // get a specific deployment
+  function get(uint _index) constant returns (address coin, address owner) {
+    Deployed deployment = deployments[_index];
+
+    coin = deployment.coin;
+    owner = deployment.owner;
+  }
+
+  // returns the number of coins for a specific owner
+  function countByOwner(address _owner) constant returns (uint) {
+    return ownedDeployments[_owner].length;
+  }
+
+  // returns a specific index by owner
+  function getByOwner(address _owner, uint _index) constant returns (address coin, address owner) {
+    uint index = ownedDeployments[_owner][_index];
+    Deployed deployment = deployments[index];
+
+    coin = deployment.coin;
+    owner = deployment.owner;
+  }
+
+  // deploy a new BasicCoin on the blockchain, optionally registering it with TokenReg
+  function deploy(uint _totalSupply, bool _withTokenreg, string _tla, string _name) payable returns (bool) {
+    Created(msg.sender, coin, _withTokenreg);
```

Agreed, however in this case we need a contract address to use it. So we cannot quite fail fast. (Dropped you a question on slack if we shouldn't just by default register and not have this option - it does cost 1ETH, only downside, but does allow us to send tokens everywhere from the app - which is the goal)